components/director/internal/domain/formationassignment/service.go

```diff
@@ -738,6 +762,12 @@ func (s *service) CleanupFormationAssignment(ctx context.Context, mappingPair *A
 		return false, nil
 	}
 
+	if response.State != nil && *response.State == string(model.DeleteErrorAssignmentState) {
```

Comment:
This `if` handles the corner case when there is an error state in the body, but no error in the response. If we don't have this `if` and there is `DELETE_ERROR` in the `response.State`, but nothing in the `response.Error` then we will not set the assignment in an error state. But we want to set it in an error state because there is a state(which is an error) in the response body and we want to respect it. Also, the message is empty, because we don't have message to put there.

PS: this was discussed with another colleague too.